src/repart/repart.c

```diff
@@ -412,6 +412,10 @@ typedef struct Partition {
         OrderedHashmap *subvolumes;
         char *default_subvolume;
         EncryptMode encrypt;
+        void *key;
+        size_t key_size;
```

Comment:
for new key, please use `struct iovec` for things like that, i.e. where you have a binary blob of some size. there are a bunch of helpers in src/basic/iovec-util.h and in src/fundamental/iovec-util-fundamental.h to make it useful.